### DISCUSSION TITLE
UCI Engineering Discussion to get clarification of APIs and workflow

### DISCUSSION THREAD

**Discussion by Nik720** [Samagra-Development/SamagraX-community#3] [Troubleshooting and Clarifications]
@SaketSinha  
{{base_url}}/admin/v1/transformer/getForms/bbf56981-b8c9-40e9-8067-468c2c753659 

The above API is returning with the list of ODK forms. From where all those ODK forms are coming?  

You have given us ODK aggregate https://agg.staging.saksham.samagra.io/Aggregate.html#management/forms where only two forms are visible, how can I get that form as a result of the above API?

what is **vaultDataMock.json** and what type of credentials this file holds? 

referring postman collection: https://documenter.getpostman.com/view/7043186/Tz5qaxaN#69b3097e-d8f5-4c9e-bad0-44350b34241e

cc @vinukumar-vs @vrayulu

**Comment by SaketSinha** [comment]
@Nik720 
ODK forms are currently coming from one of our internal ODK servers whose creds are mapped in the VaultDataMock

ValutDataMock currently is mocking the Vault (eventually, it will be updated)

I will add the ODK aggregate https://agg.staging.saksham.samagra.io/Aggregate.html#management/forms also in the vault by EOD.

**Reply by Nik720** [reply]
@SaketSinha Thank you for your response.

Please let us know once you are done with the changes, also let us know if we need to change **credentials** object values in the request body while creating an ODK transformer with a new ODK server https://agg.staging.saksham.samagra.io/

CC @vinukumar-vs @vrayulu

**Reply by Nik720** [reply]
@SaketSinha can you please help us with the below doubts

1.  Does user segments are required while creating a bot? suppose some of the ekstep scenarios do not require a specific user, it could be anonymous users then how can we create a bot? 

2. How can the user initiate a discussion with a bot? currently, what I see in bot API collection is, it uses to create a bot with conversion logic and get conversational logic.  What is the next step after getting conversion logic?  

3. How can we actually start the discussion with a bot? Can you please give us a demo with a bot to understand how can we use the APIs with a bot client?

cc @vinukumar-vs @vrayulu

**Reply by SaketSinha** [reply]
The ODK instance shared with you has been added in the vault as the default ODK setup. so all the ODK transformers you make currently will be mapped to the above mentioned odk instance.

**Comment by SaketSinha** [comment]
@Nik720 - as shared on slack please refer to this [document](https://docs.google.com/document/d/1NG2KrXwHl6vHfVz4e-niSK0PhB1UHIHkUlvWMwtYIoo/edit?usp=sharing).


This is a Step by Step guide to create a bot.

Please also refer to the FAQ section, which is divided into sections for each of the steps.

This document will eventually be made part of the wiki in a few days. Sharing it right away to unblock the team.

Do let me know if have any other questions left.

**Reply by SaketSinha** [reply]
Please confirm if this helps answer your questions.

**Reply by vinukumar-vs** [reply]
@SaketSinha Sure. We will go through the doc & confirm.

**Reply by Nik720** [reply]
@SaketSinha Thanks for sharing detailed documents, it helps a lot to understand the flow. Now we are able to test our workflow in the WhatsApp bot. can you please address the below doubts as well?
 
{{base_url}}/admin/v1/transformer/getForms/b5cb4684-e855-4c18-a7e5-a89333899ae8 
above api is not working it is giving **504 Gateway Time-out**

what I observed while testing workflow with WhatsApp bot is, every time when I add # to go to the previous step, it takes 4 minutes to get an update response

How can we integrate menu-driven workflow with portal bot client or custom bot client? In sunbird we have a separate bot client for the portal, so how can we integrate with it?

**Reply by Nik720** [reply]
@SaketSinha @ChakshuGautam 
https://agg.staging.saksham.samagra.io/Aggregate.html 
Not able to access ODK server

<img width="500" alt="Screenshot 2021-03-25 at 5 49 37 PM" src="https://user-images.githubusercontent.com/33539271/112472023-a05b1f00-8d92-11eb-9e6a-51c1d46ebd6c.png">

**Reply by SaketSinha** [reply]
@Nik720 Yes the server is down now, the team is working on it.

**Reply by ChakshuGautam** [reply]
@Nik720 I have created a separate #4 for the delay. We can discuss it there. Can you provide more information (formID) and the step at which you pressed # along with the XLSX of the form on that issue. That would enable me to reproduce it on my end. Thanks.

**Reply by SaketSinha** [reply]
FYI , the server is now up.

**Reply by aayushkedia95** [reply]
@Nik720, moved the first and third queries to separate windows.

#6 Integrating menu-driven bot with portal or custom bot client 
#7 Get Transformer Forms API giving timeout error
